### PR TITLE
Pull request for scheduling postings time and date with testing description

### PR DESCRIPTION
**Pull request by aign**
Added General Information.txt file with description of servers and connections.
Added Testing.txt file which describes testing procedures.
Corrected typo in emailer.py

### CODE REVIEW

File: sql/test_data.sql [hunk 1]
```diff
@@ -8,12 +8,14 @@ Note that both paramters to hstore should be strings and should use single quote
...
 
 INSERT INTO actions (
+	id,
 	customer_id,
 	templateurl,
 	schedule_datetime,
 	action_parameters
 	) 
 VALUES (
+	(select max(id)+1 from actions),
```
Comment (fosils) [inline sql/test_data.sql]: It is alright code but should not be done this way when using a serial. The previous code was better.

File: emailer.py [hunk 2]
```diff
@@ -130,7 +130,8 @@ def create_test_action(cur,conn):
         #create_test_action(cur,conn)
         #write_invoices_to_db(read_invoices())
         
-        cur.execute("""SELECT id,templateurl,action_parameters from actions WHERE completed is null and failed = false and ((schedule_datetime > CURRENT_TIMESTAMP) or (schedule_datetime is null))""")
+        # 
+        cur.execute("""SELECT id,templateurl,action_parameters from actions WHERE completed is null and failed = false and ((schedule_datetime < CURRENT_TIMESTAMP) or (schedule_datetime is null))""")
```
Comment (fosils) [inline emailer.py]: "#" added by mistake, I think.

File: Testing.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Testing:
+
+1. Execute AlterDataBase.sql script file to add column to table actions.
```
Comment (fosils) [inline Testing.txt]: This step is not needed for new developers, it should already have been done by you.

**Comment by fosils** [comment]
I will do the documentation myself, please try to learn from it.

File: sql/test_data.sql (hunk 1 above)
Comment (aign) [inline sql/test_data.sql]: Previous code throws error 
"ERROR:  duplicate key value violates unique constraint "actions_pkey"
DETAIL:  Key (id)=(18) already exists.
********** Error **********

ERROR: duplicate key value violates unique constraint "actions_pkey"
SQL state: 23505
Detail: Key (id)=(18) already exists." 
If there is gap in id sequence. Like  
Id
----
15
16
18
19
It inserts row with id=17, and then, next time tries to create row with id=18 , but such id already exists.

**Comment by fosils** [comment]
I have made two commits with comments:
https://github.com/OO-developers/Automatic-emailing/commit/f951b67afd96ca39d8293db55fa528646acd1f8a

and
https://github.com/OO-developers/Automatic-emailing/commit/24145b18d3974a1752064be6ae43872f48914a46

Please review them and ask if you wonder why I did it like this.

File: sql/test_data.sql (hunk 1 above)
Comment (fosils) [inline sql/test_data.sql]: If you delete all rows with id's above 17, then it should work.

Comment (aign) [inline sql/test_data.sql]: Yes. But if something goes wrong with deleting records from table (connection problems, wrong deleting script, any other problems) my code will work and will not throw error and previous code will not work at all. I have been working with databases for over 10 years (MS SQL mostly) and have seen a lot of problems with sql , and from my experience - if you can avoid script execution errors you should do it. But if you think it is better to keep previous code i will not argue.

**Comment by aign** [comment]
Ok, i get it. I used to provide documentation in in separate document.

**Comment by fosils** [comment]
Also, the most important thing to document is the purpose of a solution. Then can already see how it does things in the code, but if the developer does not know the purpose of something, then it is almost impossible to read any code.

**Comment by fosils** [comment]
Please update your pull so I can merge, and deploy your pull on the test-server.

**Pull request closed without merging**